src/dom.js

```diff
@@ -0,0 +1,371 @@
+ * rules unless exotic javascript or CSS is added after the fact.
+ *
+ * # Precise semantics
+ *
+ * The following rules are used by this function:
+ *
+ * 1) Whitespace in <pre> tags and contained tags is left alone.
+ *   In more precise terms, whitespace in any elements whose computed
+ *   `white-space` style property starts with `pre` is left alone.
+ * 2) Whitespace in other elements is compacted, meaning any combination
+ *    of whitespace characters (newlines, spaces, tabs, etc) is replaced
+ *    by a single space.
+ * 3) Any whitespace before/after closing/opening tags is removed, unless
+ *    the tag in question is inline. A tag is inline if it's computed
+ *    style property `display` starts with `inline` or is set to `content`.
+ *    This is the default behaviour for <span>.
+ * 4) Whitespace next to opening/closing tags is also collapsed; all
+ *    space between text nodes across a tree of purely inline elements are
+ *    collapsed into a single space character. The space character is placed
+ *    in the closest common ancestor, between the ancestors of both text nodes.
+ *
+ * Rule 3 and 4 are a bit verbose. Please take a look at the examples below.
+ *
+ * See also:
+ * https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace_in_the_DOM
+ * https://drafts.csswg.org/css-text-3/#propdef-white-space
+ *
+ * # Examples
+ *
+ * ` <div> </div> ` -> `<div></div>`
+ *
+ * Rule 3 - div is not inline:
+ *
+ * ` Hello <div> world </div> friend ` -> `Hello<div>world</div>friend`
+ *
+ * Rule 4 - span is inline:
+ *
+ * ` Hello <span> world </span> friend ` -> `Hello <span>world</span> friend`
+ *
+ * Rule 4 – the whitespace between multiple inline elements is placed
+ * int the lowest common ancestor.
+ *
+ * `<a>Hello   </a> \n  <a>   World</a>` -> `<a>Hello</a> <a>World</a>`
+ * `<a>Hello</a><a>   World</a>` -> `<a>Hello</a> <a>World</a>`
+ * `<span><a>Hello</a></span><a>   World</a>` -> `<span><a>Hello</a></span> <a>World</a>`
+ *
+ * # CSS Handling
+ *
+ * Note that this function does not manually check for dom nodes like
+ * <pre> or differentiate between <span> and <div>. Instead the `display`
+ * and `white-space` computed css properties are used to determine how
+ * space should be compacted.
+ *
+ * Since the computedStyle is used to determine how space compaction
+ * should be applied, the function can deal with css affecting rendering
+ * of white space: e.g. if `white-space` is set to `pre`, this will be
+ * detected by this function just as if a <pre> element had been used.
+ * The same is true for the `display` property.
+ *
+ * The only requirement for this to work is that the CSS in question is
+ * present in the dom tree.
+ *
+ * So when JSDOM is used to provide the DOM, then the entire html document
+ * should be loaded (not just fragments) and loading external stylesheets
+ * should be enabled...
+ */
+const equalizeDom = (node) =>, root = true, inlineTextNodes = []) => {
```

Comment:
- isn't here a `(` missing somewhere?
- I don't really like functions w/o a name. they don't show up in stack traces properly

Reply:
The .name property of arrow functions is defined to be the name of the variable they are assigned to; try:

```
const foo = () => {};
console.log(foo.name);
```

In general arrow functions are preferred over old style functions, because the old one follow the same weird scoping rules as the `var` statement. Try these examples:

```
console.log(foo);
const foo = () => {};
```

```
console.log(foo);
function foo() {}
```

Reply:
oh, I didn't know about the name. cool. 

```
> const foo = () => { throw Error() };
> foo()
Error
    at foo (repl:1:27)
```